src/integrationTest/java/uk/gov/hmcts/reform/wataskmanagementapi/repository/TaskResourceRepositoryTest.java

```diff
@@ -207,23 +215,173 @@ void given_task_is_created_when_find_by_id_then_task_roles_and_work_types_have_e
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId, taskIds.get(0));
+    }
+
+    @Test
+    void given_tasks_created_when_search_request_received_then_task_ids_are_returned_and_filter_by_task_type() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "IA",
+            "reviewAppeal", "anotherAssignee", "1623278362430413", CFTTaskState.COMPLETED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
+
+        Set<String> filterSignature = Set.of("*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("IA:*:*:tribunal-caseofficer:*:r:U:*", "IA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED, CFTTaskState.COMPLETED))
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal"))
+            .users(List.of("someAssignee", "anotherAssignee"))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId, taskIds.get(0));
+    }
+
+    @Test
+    void given_tasks_created_when_search_request_received_then_task_ids_are_returned_and_filter_by_assignee() {
+        String taskId2 = UUID.randomUUID().toString();
+        TaskResource createdTask = createTask(taskId2, "case-manager", "IA",
+            "reviewAppeal", "anotherAssignee", "1623278362430413", CFTTaskState.COMPLETED);
+
+        transactionHelper.doInNewTransaction(() -> {
+            task.setIndexed(true);
+            createdTask.setIndexed(true);
+            taskResourceRepository.save(task);
+            taskResourceRepository.save(createdTask);
+        });
+
+        Set<String> filterSignature = Set.of("*:IA:*:*:1:765324");
+        Set<String> roleSignature = Set.of("IA:*:*:tribunal-caseofficer:*:r:U:*", "IA:*:*:case-manager:*:r:U:*");
+        SearchRequest request = SearchRequest.builder()
+            .cftTaskStates(List.of(CFTTaskState.ASSIGNED, CFTTaskState.COMPLETED))
+            .caseIds(List.of("1623278362430412", "1623278362430413"))
+            .taskTypes(List.of("startAppeal", "reviewAppeal"))
+            .users(List.of("someAssignee"))
+            .build();
+
+        List<String> taskIds = taskResourceRepository.searchTasksIds(0,25, filterSignature, roleSignature,
+            List.of(), request);
+        assertEquals(1, taskIds.size());
+        assertEquals(taskId, taskIds.get(0));
```

Comment:
assert assignee?

Reply:
Done in RWA-2212 in CFTTaskDatabaseServiceSearchTest